metrics-jersey/src/main/java/io/stargate/metrics/jersey/ResourceTagsProvider.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.metrics.jersey;
+
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.jersey2.server.JerseyTags;
+import io.micrometer.jersey2.server.JerseyTagsProvider;
+import io.stargate.core.metrics.api.HttpMetricsTagProvider;
+import org.glassfish.jersey.server.ContainerRequest;
+import org.glassfish.jersey.server.ContainerResponse;
+import org.glassfish.jersey.server.monitoring.RequestEvent;
+
+/**
+ * Tag provider for the Jersey HTTP request.
+ *
+ * @see RequestEvent
+ * @see io.micrometer.jersey2.server.DefaultJerseyTagsProvider
+ */
+public class ResourceTagsProvider implements JerseyTagsProvider {
+
+  private final HttpMetricsTagProvider httpMetricsTagProvider;
+
+  private final Tags defaultTags;
+
+  public ResourceTagsProvider(HttpMetricsTagProvider httpMetricsTagProvider, Tags defaultTags) {
+    this.httpMetricsTagProvider = httpMetricsTagProvider;
+    this.defaultTags = defaultTags;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Iterable<Tag> httpRequestTags(RequestEvent event) {
+    // adds method, uri, status, default and request tags
+    ContainerResponse response = event.getContainerResponse();
+    ContainerRequest request = event.getContainerRequest();
+
+    Tags requestTags =
+        httpMetricsTagProvider.getRequestTags(event.getContainerRequest().getHeaders());
+
+    return defaultTags
+        .and(requestTags)
+        .and(JerseyTags.method(request), JerseyTags.uri(event), JerseyTags.status(response));
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Iterable<Tag> httpLongRequestTags(RequestEvent event) {
+    // adds method, uri, default and request tags
+    ContainerRequest containerRequest = event.getContainerRequest();
+
+    Tags requestTags =
+        httpMetricsTagProvider.getRequestTags(event.getContainerRequest().getHeaders());
+
+    return defaultTags
+        .and(requestTags)
+        .and(JerseyTags.method(containerRequest), JerseyTags.uri(event));
+  }
```

Comment:
just curious: why do we have fewer tags in the "long" version?

Reply:
Yea this actually enables you to meter the long running task before they are done.. Because of this, you can not have the status that comes from the response included.. Effectively we are not using this atm, but you could add a `@Timed(longTask = true)` to any resource method and this would kick in.. Of course this really has to be a long task, like a huge batch job..

The whole idea behind this is alerting.. if you can only measure something when it's done, then it could be already too late to alert.. This way you can set alerting and get notified once the limit is broken and "job" still runs..